From 62b1ee2aa00098a09d3c39ea7b2153e6746e0034 Mon Sep 17 00:00:00 2001
From: Patryk Kups <pkups@anaconda.com>
Date: Mon, 2 Feb 2026 09:59:30 +0100
Subject: [PATCH] patch

---
 pytest_flake8.py | 25 ++++++++++++++++++-------
 1 file changed, 18 insertions(+), 7 deletions(-)

diff --git a/pytest_flake8.py b/pytest_flake8.py
index 70b6346..d0bf4f0 100644
--- a/pytest_flake8.py
+++ b/pytest_flake8.py
@@ -4,6 +4,7 @@ import os
 import re
 from contextlib import redirect_stdout, redirect_stderr
 from io import BytesIO, TextIOWrapper
+from pathlib import Path
 
 from flake8.main import application
 
@@ -59,11 +60,11 @@ def pytest_configure(config):
             config._flake8mtimes = config.cache.get(HISTKEY, {})
 
 
-def pytest_collect_file(file_path, path, parent):
+def pytest_collect_file(file_path, parent):
     """Filter files down to which ones should be checked."""
     config = parent.config
     if config.option.flake8 and file_path.suffix in config._flake8exts:
-        flake8ignore = config._flake8ignore(path)
+        flake8ignore = config._flake8ignore(file_path)
         if flake8ignore is not None:
             item = Flake8File.from_parent(
                 parent,
@@ -201,13 +202,23 @@ class Ignorer:
             ignores.append((glob, ign))
 
     def __call__(self, path):
-        l = []  # noqa: E741
+        ignored = []  # noqa: E741
+
         for glob, ignlist in self.ignores:
-            if not glob or path.fnmatch(glob):
-                if ignlist is None:
+            if not glob:
+                match = True
+            elif isinstance(path, Path):
+                match = path.match(glob)
+            else:
+                # py.path.local
+                match = path.fnmatch(glob)
+
+            if match:
+                if ignlist is None:   # ALL
                     return None
-                l.extend(ignlist)
-        return l
+                ignored.extend(ignlist)
+
+        return ignored
 
 
 def check_file(
-- 
2.39.5 (Apple Git-154)

